### ISSUE TITLE
RRateLimiter trySetRate and expire method not an atomic method

### ISSUE THREAD

**Issue by Noel443** [redisson/redisson#6200]
When implementing rate limiting for user requests or third-party calls, we often limit based on the user's ID. However, the two operations—setting the rate limiter and setting the expiration time—aren't atomic. If the rate limiter is successfully set, but the expiration time fails to be set, it can lead to significant memory leaks. Ideally, there should be a way to ensure these operations are atomic to prevent such issues.

While I could implement this through Lua scripts, it would be much more desirable to have this functionality natively supported, as many developers encounter this scenario and would benefit from a streamlined, built-in solution. It would not only simplify development but also minimize the potential for errors.

In short, a native, atomic method for setting both the rate limiter and its expiration time would be a game-changer for managing rate limiting more efficiently and safely.

**Comment by Noel443** [comment]
When the rate limiter is configured with an expiration time of 3 minutes and a token bucket that allows 5 requests per minute, the following scenario can lead to uneven traffic distribution: If there is one request at the 1st second, followed by 4 requests at the 2nd minute and 59th second, and then another 4 requests right after the 3rd minute mark (i.e., at the 3rd minute and 3rd second), this pattern could cause the rate limiter to improperly distribute the allowed requests, potentially leading to bursts of traffic that exceed the intended rate limit.

My preferred lua code is:
```java
"redis.call('hsetnx', KEYS[1], 'rate', ARGV[1]);"
+ "redis.call('hsetnx', KEYS[1], 'interval', ARGV[2]);"
+ "local res = redis.call('hsetnx', KEYS[1], 'type', ARGV[3]);"
+ "if tonumber(ARGV[4]) > 0 then "
+ "redis.call('pexpire', KEYS[1], ARGV[4]); "
+ "end; "
+ "return res;"
```

Since I just want to keep atomicity, hsetnx will fail when the key is present, so I can change it this way to keep atomicity and refresh time. The above problem does not occur.

My business code is used like this⬇️
![image](https://github.com/user-attachments/assets/624caf1e-d5d5-4655-87cc-bc877b2f1b89)
